### PR TITLE
network-ng DNS module cleanup

### PR DESCRIPTION
**Pull request by imobachgs**
This PR contains some fixes to make the DNS UI work. Additionally, it removes some unused code.

### CODE REVIEW

File: src/include/network/services/dns.rb [hunk 1]
```diff
@@ -275,14 +275,18 @@ def StoreSettings(settings)
       )
 
       DNS.hostname = Ops.get_string(settings, "HOSTNAME", "")
-      DNS.nameservers = NonEmpty(nameservers)
+      valid_nameservers = NonEmpty(nameservers).each_with_object([]) do |ip_str, all|
+        begin
+          ip = IPAddr.new(ip_str)
+          all << ip
```
Comment (mchf) [inline src/include/network/services/dns.rb]: we already have IP validators in yast - see ```Yast::IP.Check```

File: src/lib/network/network_autoconfiguration.rb [hunk 2]
```diff
@@ -90,15 +90,7 @@ def configure_virtuals
     # Propose DNS and Hostname setup
     def configure_dns
```
Comment (mchf) [inline src/lib/network/network_autoconfiguration.rb]: it seems to be completely useless now ;-) However I'm not sure if removing ```ProposeHostname``` is correct

File: src/lib/y2network/dns.rb [hunk 3]
```diff
@@ -21,19 +21,19 @@ module Y2Network
   # DNS configuration (hostname, nameservers, etc.).
   class DNS
     # @return [String] Hostname (local part)
-    attr_reader :hostname
+    attr_accessor :hostname
```
Comment (mchf) [inline src/lib/y2network/dns.rb]: I don't think that hostname belongs here. I know that in old module it was in DNS module, however this hostname (/etc/hostname) is not much related to networking and DNS.

Comment (imobachgs) [reply src/lib/y2network/dns.rb]: We already had a discussion about that and @teclator and I did not find a better place (perhaps directly under `Y2Network::Config`), so we decided to keep it here meanwhile. But I do not have a strong opinion about moving it to a different place.

File: src/lib/network/network_autoconfiguration.rb (hunk 2 above)
Comment (imobachgs) [reply src/lib/network/network_autoconfiguration.rb]: Why?

File: src/lib/y2network/dns.rb (hunk 3 above)
Comment (teclator) [reply src/lib/y2network/dns.rb]: @mchf If I'm not wrong it was proposed to remove the handling of the `/etc/hostname` but not sure if it was really discussed or not and the conclusions as at least that option should be moved to other component.

So maybe the tab should just take care of DNS, and the `dhcp_hostname` option should be moved elsewhere (maybe in the General tab with the **DHCLIENT_SET_DEFAULT_ROUTE**)
I'f

**Comment by teclator** [comment]
For me looks good.

Maybe I would modify the validation of the `Hostname` Input in include/network/services/dns.rb. It validates it is a short Hostname, but in case the /etc/hostname is a `FQDN` it will complain and will obligate the user to modify it to apply changes in the DNS section. 

But from other POV that is also what was somehow requested in https://bugzilla.suse.com/show_bug.cgi?id=972463#c25

File: src/lib/network/network_autoconfiguration.rb (hunk 2 above)
Comment (imobachgs) [reply src/lib/network/network_autoconfiguration.rb]: I've restored the method (now it is called #propose_hostname) so we are hopefully on the safe side :)

File: src/lib/y2network/dns.rb (hunk 3 above)
Comment (mchf) [reply src/lib/y2network/dns.rb]: I was not talking about UI but the class responsibilities. If the class is named DNS then it should take care of DNS and not implement everything what was in the same module in the old code.

**Comment by mchf** [comment]
it lgtm except one np comment

**Comment by imobachgs** [comment]
> it lgtm except one np comment

It is a valid concern. We can address it when the hostname handling is finally clarified (@teclator si working on that).

**Pull request merged**